IRremoteESP8266.cpp

```diff
@@ -860,6 +860,85 @@ void ICACHE_FLASH_ATTR IRsend::sendDISH(unsigned long long data,
   }
 }
 
+// Construct a Sanyo LC7461 message.
+//
+// Args:
+//   address: The 13 bit value of the address(Custom) portion of the protocol.
+//   command: The 8 bit value of the command(Key) portion of the protocol.
+// Returns:
+//   An unsigned long long with the encoded raw 42 bit Sanyo LC7461 data value.
+//
+// Notes:
+//   This protocol uses the NEC protocol timings. However, data is
+//   formatted as : address(13 bits), !address, command (8 bits), !command.
+//   According with LIRC, this protocol is used on Sanyo, Aiwa and Chinon
+unsigned long long encodeSanyoLC7461(unsigned int address,
+                                     unsigned int command) {
+  // Mask our input values to ensure the correct bit sizes.
+  address &= SANYO_LC7461_ADDRESS_MASK;
+  command &= SANYO_LC7461_COMMAND_MASK;
+
+  unsigned long long data = address;
+  address ^= SANYO_LC7461_ADDRESS_MASK;  // Invert the 13 LSBs.
+  // Append the now inverted address.
+  data = (data << SANYO_LC7461_ADDRESS_BITS) | address;
+  // Append the command.
+  data = (data << SANYO_LC7461_COMMAND_BITS) | command;
+  command ^= SANYO_LC7461_COMMAND_MASK;  // Invert the command.
+  // Append the now inverted command.
+  data = (data << SANYO_LC7461_COMMAND_BITS) | command;
+
+  return data;
+}
+
+// Send a Sanyo LC7461 message.
+//
+// Args:
+//   data:   The contents of the command you want to send.
+//   nbits:  The bit size of the command being sent.
+//   repeat: The number of times you want the command to be repeated.
+//
+// Status: ALPHA / untested.
+//
+// Notes:
+//   Based on @marcosamarinho's work.
+//   This protocol uses the NEC protocol timings. However, data is
+//   formatted as : address(13 bits), !address, command (8 bits), !command.
+//   According with LIRC, this protocol is used on Sanyo, Aiwa and Chinon
+//   Information for this protocol is available at the Sanyo LC7461 datasheet.
+// Ref:
+//   https://github.com/marcosamarinho/IRremoteESP8266/blob/master/ir_Sanyo.cpp
+//   http://pdf.datasheetcatalog.com/datasheet/sanyo/LC7461.pdf
+void IRsend::sendSanyoLC7461(unsigned long long data, unsigned int nbits,
+                             unsigned int repeat) {
+  // Set 38kHz IR carrier frequency & a 1/3 (33%) duty cycle.
+  enableIROut(38, 33);
+  IRtimer usecTimer = IRtimer();
+
+  // Header
+  mark(SANYO_LC7461_HDR_MARK);
+  space(SANYO_LC7461_HDR_SPACE);
+  // Data
+  sendData(SANYO_LC7461_BIT_MARK, SANYO_LC7461_ONE_SPACE,
+           SANYO_LC7461_BIT_MARK, SANYO_LC7461_ZERO_SPACE, data, nbits, true);
+  // Footer
+  mark(SANYO_LC7461_BIT_MARK);
+  space(max(SANYO_LC7461_MIN_COMMAND_LENGTH - usecTimer.elapsed(),
+            SANYO_LC7461_MIN_GAP));
+
+  // Repeat
+  for (uint16_t i = 0; i < repeat; i++) {
```

Comment:
No, from what I could find out it has a similar 'repeat' method as does NEC and a few other protocols.
In that it sends a _repeat code_ to indicate a repeat, not actually repeat the message.
The fact that you questioned it means I should put some (more) comments in to indicate it's behaviour.
I'll do that now.